src/core/tsi/alts/handshaker/alts_handshaker_client.cc

```diff
@@ -39,8 +40,18 @@ struct alts_handshaker_client {
   const alts_handshaker_client_vtable* vtable;
 };
 
+struct recv_message_result {
+  tsi_result status;
+  const unsigned char* bytes_to_send;
+  size_t bytes_to_send_size;
+  tsi_handshaker_result* result;
+};
+
 typedef struct alts_grpc_handshaker_client {
   alts_handshaker_client base;
+  /* One ref is held by the entity that created this handshaker_client, and
+   * another ref is held by the pending RECEIVE_STATUS_ON_CLIENT op. */
+  gpr_refcount refs;
```

Comment:
So I believe by having this `refs`, we can correctly handle the situation when `alts_tsi_handshaker` gets destroyed while we still have a pending `RECEIVE_STATUS_ON_CLIENT` op?

Reply:
Exactly, correct